Source/Diagnostics/ComputeDiagFunctors/PhiFunctor.cpp

```diff
@@ -0,0 +1,80 @@
+/* Copyright 2025 The WarpX Community
+ *
+ * This file is part of WarpX.
+ *
+ * Authors: Roelof Groenewald (TAE Technologies)
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#include "PhiFunctor.H"
+
+#include <ablastr/constant.H>
+
+#include "WarpX.H"
+#include "Diagnostics/ComputeDiagFunctors/ComputeDiagFunctor.H"
+#include "FieldSolver/ElectrostaticSolvers/ElectrostaticSolver.H"
+
+
+using warpx::fields::FieldType;
+
+PhiFunctor::PhiFunctor (const int lev,
+    const amrex::IntVect crse_ratio,
+    bool convertRZmodes2cartesian, const int ncomp)
+    : ComputeDiagFunctor(ncomp, crse_ratio),
+    m_lev(lev), m_convertRZmodes2cartesian(convertRZmodes2cartesian)
+{ }
+
+void
+PhiFunctor::operator() (amrex::MultiFab& mf_dst, int dcomp, const int /*i_buffer*/) const
+{
+    auto& warpx = WarpX::GetInstance();
+
+    // check if phi_fp exists in the multifab registry
+    if (warpx.m_fields.has(FieldType::phi_fp, m_lev)) {
+        // grab the global phi multifab
+        amrex::MultiFab* global_phi = warpx.m_fields.get(FieldType::phi_fp, m_lev);
+
+        InterpolateMFForDiag(
+            mf_dst, *global_phi, dcomp, warpx.DistributionMap(m_lev),
+            m_convertRZmodes2cartesian
+        );
+    }
+    else
+    {
+        const int num_levels = 1;
+        // allocate fields for charge and potential
+        amrex::Vector<std::unique_ptr<amrex::MultiFab>> rho_vec(num_levels);
+        amrex::Vector<std::unique_ptr<amrex::MultiFab>> phi_vec(num_levels);
+
+        // calculate the total charge density as the divergence of the E-field
+        amrex::BoxArray ba = warpx.boxArray(m_lev);
+        ba.surroundingNodes();
+        rho_vec[0] = std::make_unique<amrex::MultiFab>(ba, warpx.DistributionMap(m_lev), 1, 1);
```

Comment:
Similarly to the comment about the `computePhi` call, it may be worth using temporary integers with variable names that explain what they are.

In this case, for example, even an IDE (or at least my IDE) doesn't help too well, because of the `std::make_unique` wrapping. 

I think the first `1` would be the `ncomp` argument and the second `1` would be the `ngrow` argument in this case.

So maybe either this
```cpp
        int const ncomp = 1;
        int const ngrow = 1;
        rho_vec[0] = std::make_unique<amrex::MultiFab>(ba, warpx.DistributionMap(m_lev), ncomp, ngrow);
```
or this
```cpp
        rho_vec[0] = std::make_unique<amrex::MultiFab>(ba, warpx.DistributionMap(m_lev), /*ncomp=*/1, /*ngrow=*/1);
```

What do you think?